.github/workflows/make_binaries.yaml

```diff
@@ -27,19 +27,34 @@ jobs:
       - name: Build PlatformIO Project
         run: pio run
 
+      - name: Copy binaries to /firmware
+        run: bash copy_firmware.sh
+
+      - name: Upload binaries for use in release step if needed
+        uses: actions/upload-artifact@v3
+        with:
+          name: firmware
+          path: firmware
+
 
   deploy_release:
     if: github.ref == 'refs/heads/main'
     runs-on: ubuntu-latest
     needs: build
     steps:
+      - uses: actions/checkout@v3
 
-      - name: Copy binaries to /firmware
-        run: bash copy_firmware.sh
+      - name: Download binaries
+        uses: actions/download-artifact@v3
+        with:
+          name: firmware
+
+      - name: List workspace directory
+        run: ls -la
 
-      - name: Create release and upload binary
+      - name: Create release and upload binaries
         run: |
-          gh release create v0.0.${{github.run_number}} firmware/*
+          gh release create v0.0.${{github.run_number}} *.bin
```

Comment:
Same comment as the workflow-update PR, but does this work? Are the files not in the subdirectory?